styx-api-service/src/test/java/com/spotify/styx/api/WorkflowResourceTest.java

```diff
@@ -583,6 +596,35 @@ public void shouldUpdateWorkflow() throws Exception {
     assertThat(deserialize(response.payload().get(), Workflow.class), equalTo(WORKFLOW));
   }
 
+  @Test
+  public void shouldFailToUpdateWorkflowIfNotAuthenticated() throws Exception {
+    sinceVersion(Api.Version.V3);
+
+    when(requestAuthenticator.authenticate(any())).thenReturn(Optional::empty);
+
+    final Response<ByteString> response = awaitResponse(
+        serviceHelper.request("POST", path("/foo"), serialize(WORKFLOW_CONFIGURATION)));
+    assertThat(response.status().family(), is(SERVER_ERROR));
```

Comment:
Why would it be `SERVER_ERROR`?

Reply:
OK I think I understand. But still confusing when reading the test case.

Reply:
Maybe we can change the test case name to `shouldHaveAssertionFailureIfMissingAuthContext`?